iceoryx_utils/include/iceoryx_utils/cxx/forward_list.hpp

```diff
@@ -0,0 +1,375 @@
+
+      private:
+        using clistNodePointer = const ListNodeType*;
+        friend class forward_list<T, CAPACITY>;
+        const forward_list<T, CAPACITY>* m_cList;
+        sizeType m_iterListNodeIdx;
+
+    }; // class const_iterator
+
+    /// @brief retrieve an interator before first element
+    ///         only allowed for usage in erase_after, insert_after, emplace_after
+    ///         Terminated when content is attemted to read (operator*, operator->)
+    /// @return iterator to meta-element before first data element
+    iterator before_begin() noexcept;
+
+    /// @brief retrieve an const_iterator before first element
+    ///         only allowed for usage in erase_after, insert_after, emplace_after
+    /// @return iterator to meta-element before first data element
+    const_iterator before_begin() const noexcept;
+
+    /// @brief const_iterator an interator before first element
+    ///         only allowed for usage in erase_after, insert_after, emplace_after
+    /// @return iterator to meta-element before first data element
+    const_iterator cbefore_begin() const noexcept;
+
+    /// @brief default list operation to retrieve an interator to first list element
+    /// @return iterator to first list element, returns iterator to end() when list is empty
+    iterator begin() noexcept;
+
+    /// @brief default list operation to retrieve an const_iterator to first list element
+    /// @return iterator to first list element, returns iterator to end() when list is empty
+    const_iterator begin() const noexcept;
+
+    /// @brief default list operation to retrieve an const_iterator to first list element
+    /// @return iterator to first list element, returns iterator to end() when list is empty
+    const_iterator cbegin() const noexcept;
+
+    /// @brief default list operation to retrieve an interator to end of list (behind last valid element)
+    ///         Terminated when content is attemted to read (operator*, operator->)
+    /// @return iterator to end element, does not contain data.
+    iterator end() noexcept;
+
+    /// @brief default list operation to retrieve an const_iterator to end of list (behind last valid element)
+    ///         Terminated when content is attemted to read (operator*, operator->)
+    /// @return iterator to end element, does not contain data.
+    const_iterator end() const noexcept;
+
+    /// @brief default list operation to retrieve an const_iterator to end of list (behind last valid element)
+    ///         Terminated when content is attemted to read (operator*, operator->)
+    /// @return iterator to end element, does not contain data.
+    const_iterator cend() const noexcept;
+
+    /// @brief list meta information on filling
+    /// @return no elements in list (true), otherwise (false)
+    bool empty() const noexcept;
+
+    /// @brief list meta information on filling
+    /// @return list is filled with 'capacity' / 'max_size' elements (true), otherwise (false)
```

Comment:
This description is somehow confusing. I would write: Returns true if list is full, otherwise false.